### ISSUE TITLE
Can we do compression based on the pretrained weight?

### ISSUE THREAD

**Issue by twmht** [microsoft/nni#3007]
**Environment**:
- NNI version: v2.0
- Python version:  3.7
- PyTorch/TensorFlow version: 1.6

**What issue meet, what's expected?**:

I found out that nni compression can't exploit a pretrained weight.

when calling `pruner.compress`, the pretrained weight will be meaningless (it seems like all the weight are all reinitialized). and the testing accuracy would be very low.

**How to reproduce it?**: 

take slim-pytorch-cifar10 (https://github.com/microsoft/nni/blob/master/examples/model_compress/slim_torch_cifar10.py) as an example

```
model = VGG(depth=19)
model.load_state_dict(torch.load('vgg19_cifar10.pth'))
model.to(device)
test(model, device, test_loader)
# top1 = 93.60%
# Pruning Configuration, in paper 'Learning efficient convolutional networks through network slimming',
configure_list = [{
        'sparsity': 0.7,
        'op_types': ['BatchNorm2d'],
}]
    
 # Prune model and test accuracy without fine tuning.
print('=' * 10 + 'Test the pruned model before fine tune' + '=' * 10)
optimizer_finetune = torch.optim.SGD(model.parameters(), lr=0.001, momentum=0.9, weight_decay=1e-4)
pruner = SlimPruner(model, configure_list, optimizer_finetune)
model = pruner.compress()
test(model, device, test_loader)
# top1 = 5%
```

i believe that this is a critical issue since using pretrained weight is very common when training object detection.

the expected testing accuracy should be same as the former ones  after  `pruner.compress`.

**Additional information**:

**Comment by twmht** [comment]
just found out the reason.

**Comment by OberstWB** [comment]
> just found out the reason.

how did you solve it?I met the same problem
